packages/waku/src/router/create-pages.ts

```diff
@@ -147,6 +147,18 @@ export type CreateLayout = <Path extends string>(
       },
 ) => void;
 
+type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';
+
+export type CreateApi = <Path extends string>(params: {
+  path: Path;
+  render: 'static' | 'dynamic';
+  method: Method;
+  handler: (params: {
+    body: ReadableStream | null;
+    headers: Readonly<Record<string, string>>;
+  }) => Promise<unknown>;
```

Comment:
As we discussed internally, this should be:
```suggestion
  handler: (req: Request) => Promise<Response>;
```

Reply:
There's a type error if I use the full `Request`, so I re-used the type from defineRouter for req

Reply:
Please create `Request` object.
```js
const convertToRequest = ({ body, headers, method }) => {
  return new Request(...);
}

Reply:
updated